theano/tensor/signal/pool.py

```diff
@@ -262,46 +322,42 @@ def __init__(self, ds, ignore_border=False, st=None, padding=(0, 0),
 
     def make_node(self, x):
         # TODO: consider restricting the dtype?
+        nd = len(self.ds)
         x = tensor.as_tensor_variable(x)
-        if x.type.ndim != 4:
+        if x.type.ndim < nd + 1:
             raise TypeError()
         # If the input shape are broadcastable we can have 0 in the output shape
-        broad = x.broadcastable[:2] + (False, False)
+        broad = x.broadcastable[:-nd] + (False,) * nd
         out = tensor.TensorType(x.dtype, broad)
         return gof.Apply(self, [x], [out()])
 
     def perform(self, node, inp, out):
         x, = inp
         z, = out
-        if len(x.shape) != 4:
+        nd = len(self.ds)
+        if len(x.shape) < nd + 1:
             raise NotImplementedError(
-                'Pool requires 4D input for now')
+                'Pool requires input with {} or more dimensions'.format(nd + 1))
```

Comment:
The op only support 4 or 5 inputs. Why you write that it support more like 6 dimensions? The pool_2d/3d interface support more, but it collapse some outer dimensions and reuse the 2d/3d pool op.